client-js/app/elements/labrad-plot.ts

```diff
@@ -154,49 +240,129 @@ export class Plot extends polymer.Base {
     for (let row of data) {
       this.splice('data', this.data.length, 0, row);
     }
-    this.plotData(data, lastData);
+    this.plotData(data);
+
+    if (!this.haveZoomed) {
+      this.resetZoom();
+    } else {
+      this.projectGraphPositions();
+    }
   }
 
 
-  private createPlot(
-      area: HTMLElement, totWidth: number, totHeight: number): void {
+  /**
+   * Redraw the plot and attach resize to the window resize event.
+   * Fires when the component is attached to the DOM.
+   */
+  attached() {
+    this.render();
+    window.addEventListener('resize', (e) => this.resizePlot());
+  }
+
+
+  /**
+   * Resets the current zoom level to fit the data.
+   */
+  resetZoomControl() {
```

Comment:
Ahh, I must have merged it poorly. The intent was to have `resetZoom` be private, then have `resetZoomControl` be the pubic hook the UI uses. 

It's basically a separation of the API the UI calls and the API the internal module calls.

Ultimately, it's a bit of a waste since it's just an alias, but the idea would be that the UI hooks can be altered independently, as well as having UI elements call descriptive names from the HTML, while internally calling a UI independent name.